keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -0,0 +1,959 @@
+
+**How**: Using a CLI in both local development and CI environments to check for breaking changes in CRD schemas
+
+#### Story 4
+
+*As a Cluster Administrator, I want to validate that updating a CustomResourceDefinition won't break my cluster*
+
+**How**: Using a CLI in both local development and CI environments to check for breaking changes in CRD schemas
+
+#### Story 5
+
+*As a Kubernetes Package Manager maintainer, I want to identify when making updates to CustomResourceDefinition(s) may result in a breaking change so that we can alert users before making those changes*
+
+**How**: Using a library for identifying breaking changes in CRD schemas
+
+#### Story 6
+
+*As a Cluster Administrator, I want Kubernetes to prevent me from making breaking changes to CRDs on my cluster*
+
+**How**: Using an admission controller/plugin that rejects CRD update operations if it would result in a breaking change
+
+**Note for reviewers**: Adding an admission controller/plugin to enact this behavior is not part of this enhancement, but it is a use case that could be enabled by the introduction of the projects being proposed.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Any practices that are codified into tooling like this are likely to influence design of APIs across the Kubernetes ecosystem to include these practices. This includes any mistakes that make it into these tools.
+
+To mitigate the risk of having a negative impact to the Kubernetes ecosystem, it is important to have each practice that is codified into these tools thoroughly reviewed and agreed upon by stakeholders prior to creating any releases that include them. Some important stakeholders to have involved in the process would be:
+- Developers creating CRDs and Kubernetes-APIs backed by Go implementations
+- Kubernetes Package Manager project maintainers (Helm, Operator Lifecycle Manager, etc.)
+- Cluster administrators (where it makes sense)
+- sig-api-machinery tech leads
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Go-based API Linting
```

Comment:
We probably want to list additional rules we want to include and why, for both this and the change validation

Reply:
https://github.com/kubernetes/enhancements/pull/5102/commits/98cd745f824da79da624e9ebeec9854114fb85bf